scripts/common.py

```diff
@@ -625,6 +633,32 @@ def wait_for_port_to_be_open(port_number):
         sys.exit(1)
 
 
+def start_elasticsearch_dev_server():
```

Comment:
Should we use the context switchers, similar to what @brianrodri used in #11510?

Reply:
Yes please, see how #11555 adds a server to ensure that the server is managed properly.

Reply:
Just updated this to use the context manager instead! PTAL

Reply:
I might still have to make a few changes to get backend tests to pass

Reply:
Sounds good, please note that because of #11549 you'll need to place `import contextlib2` lines inside of functions instead of at the top-level.